### PR TITLE
Device streaming

### PR DESCRIPTION
**Pull request by ewertons**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.
DeviceStreaming branch

## Description of the changes
Please see description of commits.

### CODE REVIEW

File: e2e/test/DeviceStreamingE2ETests.cs [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using ClientDeviceStreamingRequest = Microsoft.Azure.Devices.Client.DeviceStreamRequest;
+using ServiceDeviceStreamingRequest = Microsoft.Azure.Devices.DeviceStreamRequest;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public partial class DeviceStreamingTests : IDisposable
```
Comment (timtay-microsoft) [inline e2e/test/DeviceStreamingE2ETests.cs]: >DeviceStreamingTests [](start = 25, length = 20)

These are well written e2e tests. Good job here.

**Comment by ewertons** [comment]
This will be resubmitted.

File: e2e/test/DeviceStreamingE2ETests.cs [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+        [TestMethod]
+        public async Task ModuleStreaming_RequestRejected_Sas_Mqtt()
+        {
+            Client.Transport.Mqtt.MqttTransportSettings mqttTransportSettings =
+                new Client.Transport.Mqtt.MqttTransportSettings(Client.TransportType.Mqtt_Tcp_Only);
+            ITransportSettings[] transportSettings = new ITransportSettings[] { mqttTransportSettings };
+
+            await TestModuleStreamingAsync(TestDeviceType.Sasl, transportSettings, false).ConfigureAwait(false);
+        }
+#endregion Module Client Tests
```
Comment (prmathur-microsoft) [inline e2e/test/DeviceStreamingE2ETests.cs]: Why is commented code checked in ? :)

**Pull request closed without merging**